cmd/cuebe/cmd/delete.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright © 2021 Loft Orbital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"strings"
+
+	"cuelang.org/go/cue"
+	"github.com/loft-orbital/cuebe/cmd/cuebe/factory"
+	"github.com/loft-orbital/cuebe/cmd/cuebe/prompt"
+	"github.com/loft-orbital/cuebe/pkg/instance"
+	"github.com/spf13/cobra"
+)
+
+func newDeleteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:        "delete",
+		SuggestFor: []string{"remove", "uninstall"},
+		Short:      "Delete context from k8s cluster.",
+		Long: `
+Delete context from k8s cluster.
+
+It first group manifests found in the context by instance.
+Then it deletes those instances.
+Cuebe delete respects the deletion policy annotation "cuebe.loft-orbital.com/deletion-policy".
+		`,
+		Example: `
+# Delete all instances in the current dir
+cuebe delete .
+
+# Same but doing a dry-run
+cuebe delete --dry-run .
+`,
+		Run: runDelete,
+	}
+
+	factory.MetaOptionsAware(cmd)
+	factory.BuildAware(cmd)
+	factory.BuildContextAware(cmd)
+
+	f := cmd.Flags()
+	f.StringP("cluster", "c", "", "Cluster context. If starting with a $, it will be extracted at this path.")
```

Comment:
If by "Cluster context", you mean "kubectl config context", I suggest mentioning it cause we are jungle with different kind of "context" here and it might be confusing.